### PR TITLE
Bring verbosity improvements over from gen-code branch

### PR DESCRIPTION
**Pull request by jcburley**
This is not strictly necessary, but I found it invaluable, mainly when adding tests for verbosity levels greater than 1 (the default for one specification of `--verbose`) that would use `go-spew` to output substantial data sets.

### CODE REVIEW

File: core/data/core.joke [hunk 1]
```diff
@@ -3851,6 +3854,33 @@
...
+(add-doc-and-meta *verbose*
+  "Whether --verbose (or equivalent) was specified by the Joker invocation.
+
+  Note: changing this (via e.g. var-set) affects only running Joker code; it
+  does not change the value used by the Go code that implements Joker. If
+  needed, it's probably better to add a set-verbose function that changes
+  both, rather than changing Joker's internals so this variable is always checked."
+  {:added "1.0"})
+
+(add-doc-and-meta *verbosity-level*
```
Comment (candid82) [inline core/data/core.joke]: Having both `*verbose*` and `*verbosity-level*` seems redundant to me as `*verbosity-level*` provides a superset of possible options compared to `*verbose*`. Also, I am not sure these need to be exposed as vars. I would rather just have private `(verbosity-level__)` function that just returns `VerbosityLevel` value.

Comment (jcburley) [reply core/data/core.joke]: > Having both `*verbose*` and `*verbosity-level*` seems redundant to me as `*verbosity-level*` provides a superset of possible options compared to `*verbose*`. Also, I am not sure these need to be exposed as vars. I would rather just have private `(verbosity-level__)` function that just returns `VerbosityLevel` value.

Sounds good. I tend to default to more flexibility, but in this case it's not really warranted and actually (per the docstring) confusing. Will update PR as you suggest.

File: core/environment.go [hunk 2]
```diff
@@ -28,6 +27,8 @@ type (
 		MainFile      *Var
 		args          *Var
 		classPath     *Var
+		verbose       *Var
```
Comment (candid82) [inline core/environment.go]: This var is not used anymore and can be deleted, right?

**Comment by jcburley** [comment]
Almost certainly! Not able to do this myself just now, but feel free to fix it yourself if you like, or I'll get to it later.

On February 2, 2020 4:19:31 PM EST, Roman Bataev <notifications@github.com> wrote:
>candid82 commented on this pull request.
>
>
>
>> @@ -28,6 +27,8 @@ type (
> 		MainFile      *Var
> 		args          *Var
> 		classPath     *Var
>+		verbose       *Var
>
>This var is not used anymore and can be deleted, right?
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/candid82/joker/pull/333#pullrequestreview-351968156

-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

File: core/environment.go (hunk 2 above)
Comment (jcburley) [reply core/environment.go]: No, and neither is `verbosity` needed, nor the code that initializes them.

Gone!

**Comment by candid82** [comment]
Thanks a lot!

**Pull request merged**